arch/arm64/boot/dts/qcom/sdm636-bbry-luna.dts

```diff
@@ -0,0 +1,761 @@
+			size = <0x00 0x800000>;
+		};
+
+		adsp_fw_region {
+			compatible = "shared-dma-pool";
+			alloc-ranges = <0x00 0x00 0x00 0xffffffff>;
+			reusable;
+			alignment = <0x00 0x400000>;
+			size = <0x00 0x800000>;
+		};
+
+		qseecom_region {
+			compatible = "shared-dma-pool";
+			alloc-ranges = <0x00 0x00 0x00 0xffffffff>;
+			reusable;
+			alignment = <0x00 0x400000>;
+			size = <0x00 0x1c00000>;
+		};
+
+		secure_region {
+			compatible = "shared-dma-pool";
+			alloc-ranges = <0x00 0x00 0x00 0xffffffff>;
+			reusable;
+			alignment = <0x00 0x400000>;
+			size = <0x00 0x5c00000>;
+		};
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			alloc-ranges = <0x00 0x00 0x00 0xffffffff>;
+			reusable;
+			alignment = <0x00 0x400000>;
+			size = <0x00 0x2c00000>;
+			linux,cma-default;
+		};
+
+		rmtfs_mem: memory@fdc00000 {
+			compatible = "qcom,rmtfs-mem";
+			reg = <0x0 0xfdc00000 0x0 0x400000>;
+			no-map;
+
+			qcom,client-id = <1>;
+			qcom,vmid = <QCOM_SCM_VMID_MSS_MSA>;
+		};
+	};
+	
+	vph_pwr: vph-pwr-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "vph_pwr";
+		regulator-min-microvolt = <3700000>;
+		regulator-max-microvolt = <3700000>;
+		regulator-always-on;
+		regulator-boot-on;
+	};
+
+	battery: battery {
+		compatible = "simple-battery";
+		charge-full-design-microamp-hours = <3000000>;
+		voltage-min-design-microvolt = <3400000>;
+		voltage-max-design-microvolt = <4350000>;
+	};
+
+	keypad_vdd_vreg: keypad-vdd-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "keypad_vdd_vreg";
+		gpio = <&tlmm 42 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+
+		pinctrl-0 = <&keypad_pwr_en>;
+		pinctrl-names = "default";
+	};
+};
+
+&pon_pwrkey {
+	status = "okay";
+};
+
+&pon_resin {
+	linux,code = <KEY_VOLUMEDOWN>;
+
+	status = "okay";
+};
+
+&adreno_gpu {
+	status = "okay";
+};
+
+&adreno_gpu_zap {
+	memory-region = <&zap_shader_region>;
+	firmware-name = "qcom/sdm636/luna/a512_zap.mbn";
+};
+
+&adsp_pil {
+	firmware-name = "qcom/sdm636/luna/adsp.mbn";
+};
+
+&blsp2_uart1 {
+	status = "okay";
+
+	/* HCI Bluetooth */
+	bluetooth {
+		compatible = "qcom,wcn3990-bt";
+		vddxo-supply = <&vreg_l9a_1p8>;
+		vddrf-supply = <&vreg_l6a_1p3>;
+		vddch0-supply = <&vreg_l3b_3p3>;
+		vddio-supply = <&vreg_l9a_1p8>;
+		max-speed = <3200000>;
+		firmware-name = "luna/crnv21.bin";
+	};
+};
+
+&blsp1_uart2 {
+	status = "okay";
+};
+
+
+&blsp_i2c1 {
+	status = "okay";
+
+	stmpe1801: port-expander@40
+	{
+		compatible = "st,stmpe1801";
+		reg = <0x40>;
+		vcc-supply = <&keypad_vdd_vreg>;
+		vio-supply = <&keypad_vdd_vreg>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&keypad_int_active &keypad_reset_active>;
+		wakeup-source;
+
+		interrupt-parent = <&tlmm>;
+		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
+
+		keyboard-controller {
+			compatible="st,stmpe-keypad";
+			st,scan-count = <2>;
+			debounce-interval = <10>;
+			keypad,num-rows = <8>;
+			keypad,num-columns = <8>;
+
+			linux,keymap = <
+				// Row 1
```

Comment:
C-style comments seem to be not forbidden, at least they are not mentioned in [dts-coding-style](https://github.com/sdm660-mainline/linux/blob/qcom-sdm660-6.15.y/Documentation/devicetree/bindings/dts-coding-style.rst), but all examples there use C-style `/* .. */` multiline comments everywhere. This is requirement for linux kernel C source files, not for DTS, but still...  though [devicetree-specification](https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.4) explicitly states 
```
C style (/* ... \*/) and C++ style (//) comments are supported.
```
Perhaps better to change them all to C style comments to be safe, but not a blocker here.